### DISCUSSION TITLE
rest API

### DISCUSSION THREAD

**Discussion by whateverUsernameAvailable** [mitmproxy/mitmproxy#5169] [Q&A] [answered]
Hello, I am sorry to bother you but I am unable to find any info/docs about your rest API, but I believe there is some. Would you be so kind and point me to some more info/docs to your rest API ?

Thanks.

**Comment by mhils** [comment, marked as answer]
There is no public REST API. mitmweb provides an internal API, but there are no stability guarantees or documentation. You can probably figure out most stuff by looking at https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/tools/web/app.py, but be prepared that this may break with future versions.

**Comment by hacker1024** [comment]
If you're looking to intercept and modify requests and responses remotely, I've just written an add-on that provides a WebSocket-based API for this: https://github.com/hacker1024/mitmproxy_remote_interceptions.

It's by no means perfect (I did make the thing so I _wouldn't_ have to use Python, after all), but it works for my purposes.

**Reply by hacker1024** [reply]
Oh, and here's a Dart client as well: https://github.com/hacker1024/mitmproxy_ri_client.dart
